.github/workflows/sonarcloud.yml

```
@@ -10,39 +10,73 @@ jobs:
+      - name: Set environment variables for CI
+        run: |
+          echo "CI_DATABASE_URL=postgresql://sonar:sonar@localhost:5432/sonar" >> $GITHUB_ENV
+          echo "ENV=ci" >> $GITHUB_ENV
+          echo "DB_HOST=localhost" >> $GITHUB_ENV
+          echo "DB_PORT=5432" >> $GITHUB_ENV
+          echo "DB_NAME=sonar" >> $GITHUB_ENV
+          echo "DB_USER=sonar" >> $GITHUB_ENV
+          echo "DB_PASSWORD=sonar" >> $GITHUB_ENV
+
       - name: Set up Python
         uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)